### PR TITLE
chore(deps): issue-6078, try to make goldenfile results readable

### LINKED ISSUE #6078: Test output of goldenfile is not readable
We use `goldenfile` for AST unit tests, but it will output the whole content of goldenfile while the test failed. This behavior makes it hard for debugging.

As we already print the diff, maybe we can omit the output of goldenfile's content?

image

Take `test_statement` as an example, we have to `PgUp` several times to know what happened.

@andylokandy @leiysky

### PR DESCRIPTION
**Pull request by PsiACE**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- apply my patch of rust-goldenfile
- check https://github.com/calder/rust-goldenfile/pull/7#issuecomment-1162006531 to see the display in our case

## Changelog

- Not for changelog (changelog entry is not required)

## Related Issues

Fixes #6078

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -68,3 +68,4 @@ rustc-demangle = { opt-level = 3 }
 [patch.crates-io]
 parquet2 = { version = "0.13", optional = true, git = "https://github.com/datafuse-extras/parquet2", branch = "parquet2-0.13-patch2" }
 chrono = { git = "https://github.com/datafuse-extras/chrono", rev = "279f590" }
+goldenfile = { git = "https://github.com/psiace/rust-goldenfile", rev = "16c5783" }
```
Comment (Xuanwo) [inline Cargo.toml:71]: Please add a comment with links to https://github.com/calder/rust-goldenfile/pull/7 so that we can switch back to upstream later.

**Review by Xuanwo — APPROVED** [review]

**Comment by Xuanwo** [comment]
Nice! Developers' lives will be much easier!

File: Cargo.toml [hunk 2]
```diff
@@ -68,3 +68,5 @@ rustc-demangle = { opt-level = 3 }
 [patch.crates-io]
 parquet2 = { version = "0.13", optional = true, git = "https://github.com/datafuse-extras/parquet2", branch = "parquet2-0.13-patch2" }
 chrono = { git = "https://github.com/datafuse-extras/chrono", rev = "279f590" }
+# https://github.com/calder/rust-goldenfile/pull/7
+goldenfile = { git = "https://github.com/psiace/rust-goldenfile", rev = "16c5783" }
```
Comment (BohuTANG) [inline Cargo.toml:72]: Can we use a crate, not a GitHub repo?

Comment (PsiACE) [reply Cargo.toml:72]: This is just a patch approach, and once the upstream merge is done, we can say goodbye to the github repo.

Or, if necessary, I can maintain a new crate.

Comment (Xuanwo) [reply Cargo.toml:72]: It doesn't worth implementing a new crate for such change. If upstream reject or no-responding to our PR, we can implement [`Differ`](https://docs.rs/goldenfile/latest/goldenfile/differs/type.Differ.html) instead.

Comment (Xuanwo) [reply Cargo.toml:72]: @BohuTANG Do those explanations make sense to you?  Please take a look again.

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**